tests/Doctrine/Tests/ORM/Functional/Ticket/GH5742Test.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\ORM\Mapping as ORM;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+class GH5742Test extends OrmFunctionalTestCase
+{
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->createSchemaForModels(
+            GH5742Person::class,
+            GH5742Toothbrush::class,
+            GH5742ToothpasteBrand::class
+        );
+    }
+
+    public function testUpdateOneToOneToNewEntityBeforePreviousEntityCanBeRemoved(): void
+    {
+        $this->expectNotToPerformAssertions();
+
+        $person             = new GH5742Person();
+        $oldToothbrush      = new GH5742Toothbrush();
+        $person->toothbrush = $oldToothbrush;
+
+        $this->_em->persist($person);
+        $this->_em->persist($oldToothbrush);
+        $this->_em->flush();
+
+        $newToothbrush      = new GH5742Toothbrush();
+        $person->toothbrush = $newToothbrush;
+
+        $this->_em->remove($oldToothbrush);
+        $this->_em->persist($newToothbrush);
+
+        // The flush operation will have to make sure the new toothbrush
+        // has been written to the database
+        // _before_ the person can be updated to refer to it.
+        // Likewise, the update must have happened _before_ the old
+        // toothbrush can be removed (non-nullable FK constraint).
+
+        $this->_em->flush();
+    }
+
+    public function testManyToManyCollectionUpdateBeforeRemoval(): void
+    {
+        $this->expectNotToPerformAssertions();
+
+        $person             = new GH5742Person();
+        $person->toothbrush = new GH5742Toothbrush(); // to satisfy not-null constraint
+        $this->_em->persist($person);
+
+        $oldMice = new GH5742ToothpasteBrand();
+        $this->_em->persist($oldMice);
+
+        $person->preferredBrands->set(1, $oldMice);
+        $this->_em->flush();
+
+        $newSpice = new GH5742ToothpasteBrand();
```

Comment:
"old mice"? "new spice" What's the joke? I don't get it… :thinking:

Reply:
Old mice 🐁 is not so yum.

New spice might me something fresh like peppermint?

Reply:
Makes sense :)